ckanext/cfpb_extrafields/templates/macros/form_custom.html

```diff
@@ -0,0 +1,62 @@
+{# pick-one list with specified other option #}
+{% macro select_one_with_other(name, label='', current_value="", options_defaults=[], extra_html="", error='' ) %}
+<div class="control-group{{ " error" if error }}{{ " " ~ classes | join(' ') }}">
+    <label class="control-label" for="{{ name }}">{{label}}</label>
+    <div class="controls{{ " " ~ control_classes | join(' ') }}">
+    <select id={{name}} name={{name}} width="400" style="width: 400px">
+    {# provide nothing as an option #}
+    <option value=""></option>
+    {# move previously selected elements to the top and select #}
+    {% if current_value %}<option value="{{current_value}}" selected="selected">{{current_value}}</option>{% endif %}
+    {# provide the default options but don't duplicate if they were selected above #}
+    {% for e in options_defaults %}
+        {% if e != current_value %}
+            <option value="{{e}}">{{e}}</option>
+        {% endif %}
+    {% endfor %}
+    {# provide an option for javascript to use to create new fields #}
+    <option value="__Other">Other, please specify</option>
+    </select>
+    {% if error and error is iterable %}<span class="error-block">{{ error|join(', ') }}</span>{% endif %}
```

Comment:
That's kind of my point here, those funny edge cases, if we don't show anything here when they do occur it could be problematic.  I'd recommend we either remove the "is iterable" or we handle the edge case by printing it verbatim.